### PR TITLE
Hotfix/customizer sanitization

### PR DESCRIPTION
**Pull request by praghavaji**

### CODE REVIEW

File: src/extensions/monetize-block.js [hunk 1]
```diff
@@ -170,12 +170,15 @@ const monetizeBlockControls = createHigherOrderComponent( ( BlockEdit ) => {
...
 		// Only show inspector options if set for block level monetization.
 		showInspector = false;
+		// console.log('meta', meta);
 		if ( typeof meta !== 'undefined' ) {
 			if ( typeof meta._coil_monetize_post_status === 'undefined' || ( typeof meta._coil_monetize_post_status !== 'undefined' && meta._coil_monetize_post_status === 'gate-tagged-blocks' ) ) {
 				showInspector = true;
 			}
 		}
 
+		// console.log('showInspector', showInspector);
```
Comment (paulgibbs) [inline src/extensions/monetize-block.js]: Does this debug need to stay in?

File: package.json [hunk 2]
```diff
@@ -59,6 +59,7 @@
 	"dependencies": {
 		"cgb-scripts": "1.21.0",
 		"classnames": "^2.2.6",
+		"gulp": "^4.0.2",
```
Comment (paulgibbs) [inline package.json]: What's gulp for?

**Comment by paulgibbs** [comment]
@praghavaji Is the intent of this PR to allow HTML to be entered/saved into this text field?

File: src/extensions/monetize-block.js (hunk 1 above)
Comment (praghavaji) [reply src/extensions/monetize-block.js]: Nope. Do I need to remove it to get approval?

File: package.json (hunk 2 above)
Comment (praghavaji) [reply package.json]: I have no idea. I didn't change the package.json file.

Curious.

Comment (paulgibbs) [reply package.json]: Try to merge develop branch (or whatever the main dev branch) back into your PR is, and see if it still show in the diff. If it still does, then you did change it. :)

File: src/extensions/monetize-block.js (hunk 1 above)
Comment (paulgibbs) [reply src/extensions/monetize-block.js]: It's sloppy leaving commented out code behind, so yes please, remove it. :)

Comment (praghavaji) [reply src/extensions/monetize-block.js]: Ok. Will do.

File: package.json (hunk 2 above)
Comment (praghavaji) [reply package.json]: Okey doke.

**Comment by praghavaji** [comment]
> @praghavaji Is the intent of this PR to allow HTML to be entered/saved into this text field?

Not from my point of view. The previous HTML encoding was being applied at the point of rendering and saving, thus rendering the escaped text for the user to see.

My intention here was just to properly encode and sanitize in the right places cleaning that up.

Make sense?

**Comment by praghavaji** [comment]
Hey @paulgibbs .. I;ve made those changes and pushed the cleaned-up branch. Do I need to create a new pull request for this now?

File: tests/cypress/integration/coil/single-post.js [hunk 3]
```diff
@@ -5,8 +5,8 @@ const paymentPointer = 'https://example.com/' + Math.random().toString(36) + '/.
 
 // Most of these tests assume you have the test posts loaded in your WordPress.
 describe('Single Posts', function () {
-  beforeEach(() => {
-    cy.logInToWordPress('admin', 'password');
+	  beforeEach(() => {
+	    cy.logInToWordPress('madhavaji', 'WdX2aJ36iFyZbda9');
```
Comment (paulgibbs) [inline tests/cypress/integration/coil/single-post.js:9]: Don't commit passwords!

File: includes/admin/functions.php [hunk 4]
```diff
@@ -368,7 +368,7 @@ function add_customizer_messaging_panel( $wp_customize ) : void {
 		$partial_message_id,
 		[
 			'capability'        => apply_filters( 'coil_settings_capability', 'manage_options' ),
-			'sanitize_callback' => 'wp_filter_nohtml_kses',
+			'sanitize_callback' => 'esc_textarea',
```
Comment (paulgibbs) [inline includes/admin/functions.php:371]: esc_textarea is an escaping function, not a sanitisation function. This won't be safe to use here.

You should do some research and test this but I think `sanitize_text_field` might be appropriate here.

**Review by paulgibbs — CHANGES_REQUESTED** [review]

**Comment by iamthomasbarrett** [comment]
Addressed in PR 103.

**Pull request closed without merging**